rasterio/features.py

```diff
@@ -507,7 +507,7 @@ def is_valid_geom(geom):
 
 def dataset_features(
         src,
-        bidx,
+        bidx=None,
```

Comment:
@perrygeo checking with you since this is your work... turning this into a keyword argument made `dataset_features` work better with `functools.partial`.